public/app/features/manage-dashboards/DashboardImportPage.tsx

```diff
@@ -123,20 +120,26 @@ class UnthemedDashboardImport extends PureComponent<Props> {
 
     const dashboard = JSON.parse(formData.dashboardJson);
 
+    if ((dashboard.spec?.elements || dashboard.elements) && !config.featureToggles.dashboardNewLayouts) {
```

Comment:
for one of my dashboards I tested, I went through the sequence `v1 → v2 → v1`. After switching back to v1, the dashboard ended up with an empty `"elements": {}` object. 
With `dashboardNewLayouts` enabled, this did not let me import dashboard, even though it was technically a v1 version. So should we check for `dashboard.elements` or just stick with `dashboard.spec?.elements` instead?

Also, let me know if you need the full dashboard json!